src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -349,6 +354,31 @@ abstract class FixupSession(
           }
         }
 
+        // Skip the formatting if these edits are part of an undo operation.
+        if (!undoInProgress) {
+          // It is safe to use selectionRange here, because it is set in the constructor
+          // by calling triggerFixupAsync() which in turn calls ensureSelectionRange().
+          // The value of this property does not change
+          // when the user selects a different snippet in the UI.
+          // FIXME this causes a problem when the entire file becomes shorter than the original
```

Comment:
Is that going to be fixed as part of this PR?

Reply:
I have refactored `FixupSession` to use the `RangeMarker` from the IntelliJ API. Unlike the dumb data classes such as `Range`, `TextRange` etc., the `RangeMarker` remains tied to its document, and updates itself whenever the document is modified.

This has eliminated the problems with shortening the file. However, I've run into other problems...